internal/datastore/common/gc.go

```diff
@@ -88,31 +91,46 @@ type GarbageCollector interface {
 
 	// ResetGCCompleted resets the state of the garbage collection run.
 	ResetGCCompleted()
+}
+
+// GarbageCollector represents any datastore that supports external garbage
+// collection.
+type GarbageCollector[C any] interface {
+	RunnableGarbageCollection
+
+	// AcquireConnection acquires a connection to the datastore for use by the GC
+	// process. If no dedicated connection is necessary, return nil.
+	AcquireConnection(ctx context.Context) (C, error)
```

Comment:
I think you could take C out of the interface:
```go
	AcquireConnection(ctx context.Context) error
	ReleaseConnection()
```
and instead track it as an internal member of the type that implements `GarbageCollector`

Reply:
There is no requirement that the datastore itself act as a singleton garbage collector, so strictly speaking, we can't. We could have the datastore return an instance of a GarbageCollection interface, and then we could remove the connection, but that's a different design

Reply:
Wouldn't this just not be implemented by datastores that don't need a garbage collector?